charts/ChartUrl.ts

```diff
@@ -277,15 +276,18 @@ export class ChartUrl implements ObservableUrl {
 
     setTimeFromTimeQueryParam(time: string) {
         const { chart } = this
-        // We want to support unbounded time parameters, so that time=2015.. extends from 2015
-        // to the latest year, and time=..2020 extends from earliest year to 2020. Also,
-        // time=.. extends from the earliest to latest available year.
+
+        // In the past we supported unbounded time parameters like time=2015.. which would be
```

Comment:
nit: one pattern that also might work here for maintainability is to break out the conversion of legacy time params into the current form, something like:

```typescript
legacyTimeToTimeParam(time: string) {
 if ("..latest") return (...)
}
```
And then call that method somewhere in the chain.

Then at some point we could look at analytics and if no one is using those old time params could drop that code.

Reply:
This makes sense, and I see how you are doing it in `CovidChartUrl`.
But in this case I think this over-complicates things since separating these two code paths is not that easy actually.